### ISSUE TITLE
Failing tests if wiremock::Mock is not assigned to variable

### ISSUE THREAD

**Issue by LuisBarroso37** [LukeMathWalker/zero-to-production#115]
In the function `create_unconfirmed_subscriber` located in the file `/src/tests/api/newsletter.rs`, I noticed that if you don't assign wiremock::Mock to a variable (in the book, it is assigned to `_mock_guard`), the 2 tests which call this function fail. This happens because the assertions made by wiremock::Mock are not performed.

Is this normal behaviour?

This is the code in question:
```
let _mock_guard = Mock::given(path("/email"))
        .and(method("POST"))
        .respond_with(ResponseTemplate::new(200))
        .named("Create unconfirmed subscriber")
        .expect(1)
        .mount_as_scoped(&app.email_server)
        .await;
```

**Comment by LukeMathWalker** [comment]
It is expected behaviour, but the compiler should emit a warning telling you to bound the returned guard to a variable.
I've fixed this in `wiremock` `0.5.7`!

**Comment by LuisBarroso37** [comment]
Ok, thank you! I will update wiremock to the latest version 😊
